svc-mesh/service-mesh.md

```diff
@@ -0,0 +1,229 @@
+# Kata Containers and service mesh for Kubernetes
+
+A service mesh is a powerful way to monitor and control the traffic
+between micro-services running in your Kubernetes cluster. They are
+powerful tools that you might want to use in combination with the
+security brought by Kata Containers.
+
+## How do they work ?
+
+They both rely on the same model, where they run controller applications
+in the control plane, and inject a proxy as a sidecar inside the service
+pod. The proxy registers to the control plane as a first step, and it
+constantly sends different sort of information about the service running
+inside the pod. Those information come from the filtering performed when
+receiving all the traffic initially intended for the service. That's how
+the interaction between the control plane and the proxy allows the user
+to apply load balancing and authentication rules to the on going traffic
+inside the cluster between multiple micro-services.
+
+I forgot to mention this magic could not happen without a good amount
+of `iptables` rules ensuring the packets reach the proxy instead of
+the expected service. Those rules are setup through an __init__ container
+because they have to be there as soon as the proxy starts.
+
+## Prerequisites
+
+### Kata and Kubernetes
+
+Follow the [instructions](https://github.com/kata-containers/documentation/blob/master/install/README.md)
+to get Kata Containers properly installed and configured with Kubernetes.
+You can make any choice you want between CRI-O and CRI-containerd, both
+will be supported through this document.
+
+For both cases, select the workloads as _trusted_ by default. This way,
+your cluster and your service mesh will run with `runc`, and only the
+containers you choose to annotate will run with Kata Containers.
+
+### Restrictions
+
+As documented [here](https://github.com/runconduit/conduit/issues/982),
+a kernel version between 4.14.22 and 4.14.40 will cause a deadlock when
+`getsockopt()` get called with `SO_ORIGINAL_DST` option. Unfortunately,
+both service mesh use this system call with this same option from the
+proxy container running inside the VM. This means that you cannot run
+this kernel version range as the guest kernel for Kata if you want your
+service mesh to work.
+
+As mentioned when explaining the basic functioning of those service mesh,
+`iptables` are heavily used, and they need to be properly enabled through
+the guest kernel config, otherwise the __init__ container will not be able
+to perform proper setup of the rules.
+
+## Install and deploy your service mesh
+
+### Istio
+
+As a reference, you can follow Istio [instructions](https://istio.io/docs/setup/kubernetes/quick-start/#download-and-prepare-for-the-installation).
+
+Here is a summary of what you need to install Istio 0.8.0 on your system
+```
+curl -L https://git.io/getLatestIstio | sh -
```

Comment:
Please prefix all command lines the user is expected to run with `$ ` as documented in https://github.com/kata-containers/documentation/blob/master/Documentation-Requirements.md.

Reply:
done